app/models/user.rb

```diff
@@ -22,7 +22,7 @@
 #
 
 class User < ApplicationRecord
-  ROLES = { sale: 0, admin: 1, moderator: 2 }.freeze
+  ROLES = { sale: 'sale', admin: 'admin', moderator: 'moderator' }.freeze
```

Comment:
REMOVE enum
and make
SALE_ROLE = 'sale'
ROLES  = [SALE_ROLE, MODERATOR_ROLE....]